src/v/config/configuration.cc

```diff
@@ -684,7 +684,7 @@ configuration::configuration()
       *this,
       "metadata_dissemination_retry_delay_ms",
       "Delay before retrying a topic lookup in a shard or other meta tables.",
-      {.visibility = visibility::tunable},
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
```

Comment:
As discussed in the slack thread, this change would likely require making multiple call sites dynamic. Since this config is rarely (probably never?) tuned in practice, I think it's ok to keep it as needs_restart::yes and instead update RNOT not to update it during the test run.

Reply:
I think we need to know the intention first, because right now it seems like worst case scenario: a needs_restart config is still udpated in memory, and right now it looks like the result of that is that we have part of the code use the new value and part of the code use the cached value.

It would seem like the options are:

1. change the call sites to be dynamic and leave RNOT unchanged
2. change the call sites to cache the value and remove the config from RNOT
3. change no call sites and remove the config from RNOT